Don't set hardware WEP if we are actually using TKIP/AES.

Signed-off-by: Hong Liu <hong.liu@intel.com>
--
diff -Nurp ipw2100-1.1.2/ipw2100.c ipw2100-1.1.2-wext-crash/ipw2100.c
--- ipw2100-1.1.2/ipw2100.c	2005-07-14 02:59:37.000000000 +0800
+++ ipw2100-1.1.2-wext-crash/ipw2100.c	2005-08-03 16:07:06.000000000 +0800
@@ -5186,8 +5186,11 @@ static void shim__set_security(struct ne
 			else
 				memcpy(priv->ieee->sec.keys[i], sec->keys[i],
 				       sec->key_sizes[i]);
-			priv->ieee->sec.flags |= (1 << i);
-			priv->status |= STATUS_SECURITY_UPDATED;
+			if (sec->level == SEC_LEVEL_1) {
+				priv->ieee->sec.flags |= (1 << i);
+				priv->status |= STATUS_SECURITY_UPDATED;
+			} else
+				priv->ieee->sec.flags &= ~(1 << i);
 		}
 	}
 
